browser/extensions/api/settings_private/brave_prefs_util.cc

```diff
@@ -56,7 +56,7 @@
 #endif
 
 #if BUILDFLAG(ENABLE_PLAYLIST)
-#include "brave/components/playlist/browser/pref_names.h"
+#include "brave/components/playlist/common/pref_names.h"
```

Comment:
why is this moved into common? That is only needed for code that is shared across processes. In this case it looks like we need to split playlist into content/core because browser has content deps so I would do that first and then follow-up with this PR

Reply:
@bridiver We needed to get access to this header without using the `browser` target which relies on banned iOS deps, so solutions were:
- moving it into common
- creating a separate source_set entirely for just `pref_names.h` and dep on that
- altering the main `//brave/components/playlist/browser/BUILD.gn` with a bunch of `is_ios` checks to only have pref_names sources 

The first option seemed the most common practice (even among non-content/core components)